core/src/main/java/io/kestra/core/runners/RunContext.java

```diff
@@ -2,989 +2,175 @@
-        } else {
-            logger().warn("Unable to decrypt the output as encryption is not configured");
-            return encrypted;
-        }
-    }
+    String decrypt(String encrypted) throws GeneralSecurityException;
 
     /**
      * Encrypt a plaintext string using the {@link EncryptionService} and the default encryption key.
      * If the key is not configured, it will log a WARNING and return the plaintext string as is.
      */
-    public String encrypt(String plaintext) throws GeneralSecurityException {
-        if (secretKey.isPresent()) {
-            return EncryptionService.encrypt(secretKey.get(), plaintext);
-        } else {
-            logger().warn("Unable to encrypt the output as encryption is not configured");
-            return plaintext;
-        }
-    }
-
-    public org.slf4j.Logger logger() {
-        return runContextLogger.logger();
-    }
+    String encrypt(String plaintext) throws GeneralSecurityException;
 
     /**
-     * Gets a {@link InputStream} for the given file URI.
+     * Gets the {@link Logger} attached to this {@link RunContext}.
      *
-     * @param uri   the file URI.
-     * @return      the {@link InputStream}.
-     * @throws IOException
-     * @deprecated use {@link Storage#getFile(URI)}.
+     * @return the {@link Logger}.
      */
-    @Deprecated
-    public InputStream uriToInputStream(URI uri) throws IOException {
-        return this.storage.getFile(uri);
-    }
+    Logger logger();
 
     // for serialization backward-compatibility
     @JsonIgnore
-    public URI getStorageOutputPrefix() {
-        return storage.getContextBaseURI();
-    }
+    URI getStorageOutputPrefix();
 
     /**
-     * Gets access to the Kestra's storage.
+     * Gets access to the Kestra's internal storage.
      *
-     * @return  a {@link Storage} object.
+     * @return a {@link Storage} object.
      */
-    public Storage storage() {
-        return storage;
-    }
+    Storage storage();
+
+    List<AbstractMetricEntry<?>> metrics();
+
+    <T> RunContext metric(AbstractMetricEntry<T> metricEntry);
+
+    void dynamicWorkerResult(List<WorkerTaskResult> workerTaskResults);
+
+    List<WorkerTaskResult> dynamicWorkerResults();
 
     /**
-     * Put the temporary file on storage and delete it after.
+     * Returns the local temporary directory associated to this context.
      *
-     * @param file  the temporary file to upload to storage
-     * @return      the {@code StorageObject} created
-     * @throws IOException If the temporary file can't be read
+     * <p>
+     * This method automatically creates the temporary directory if not already exists.
      *
-     * @deprecated use {@link #storage()} and {@link InternalStorage#putFile(File)}.
+     * @return  the temp-dir path.
      */
-    @Deprecated
-    public URI putTempFile(File file) throws IOException {
-        return this.storage.putFile(file);
+    default Path tempDir() {
+        return tempDir(true);
     }
```

Comment:
Is there a reason why it's the only method having a default impl?
I'd prefer to have all impl in the MutableRunContext